Source/WebCore/xml/parser/XMLDocumentParserScope.cpp

```diff
@@ -28,27 +28,33 @@
 
 namespace WebCore {
 
-CachedResourceLoader* XMLDocumentParserScope::currentCachedResourceLoader = nullptr;
+WeakPtr<CachedResourceLoader>& XMLDocumentParserScope::currentCachedResourceLoader()
+{
+    static NeverDestroyed<WeakPtr<CachedResourceLoader>> currentCachedResourceLoader;
+    return currentCachedResourceLoader;
+}
 
-XMLDocumentParserScope::XMLDocumentParserScope(CachedResourceLoader* cachedResourceLoader)
-    : m_oldCachedResourceLoader(currentCachedResourceLoader)
 #if ENABLE(XSLT)
-    , m_oldGenericErrorFunc(xmlGenericError)
-    , m_oldStructuredErrorFunc(xmlStructuredError)
-    , m_oldErrorContext(xmlGenericErrorContext)
-#endif
+XMLDocumentParserScope::XMLDocumentParserScope(WeakPtr<CachedResourceLoader> cachedResourceLoader)
```

Comment:
We shouldn't pass a WeakPtr in parameter here. The previous code matches the guidelines.

Reply:
FIXED